online/src/main/scala/ai/chronon/online/FetcherModelUtils.scala

```diff
@@ -0,0 +1,171 @@
+package ai.chronon.online
+
+import ai.chronon.api.Extensions.{ModelTransformOps, ThrowableOps}
+import ai.chronon.api.ModelTransform
+import ai.chronon.online.Fetcher.ResponseWithContext
+import ai.chronon.online.Metrics.Environment
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.Seq
+import scala.concurrent.{ExecutionContext, Future}
+
+// Unique identifier for a model request, used to join inference result back to the original join request
+case class ModelRequestIdentifier(joinName: String, modelName: String, prefix: Option[String]) {
+  def stringRep: String = (Seq(joinName) ++ prefix.toSeq ++ Seq(modelName)).mkString("_")
+}
+case class ModelTransformRequest(model: ModelTransform,
+                                 modelRequestIdentifier: ModelRequestIdentifier,
+                                 inputs: Map[String, AnyRef])
+case class ModelTransformResponse(modelRequestIdentifier: ModelRequestIdentifier,
+                                  outputs: Map[String, AnyRef],
+                                  exceptions: Map[String, Throwable])
+
+object FetcherModelUtils {
+
+  @transient implicit lazy val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  /*
+   * Convert each join request into a list of ModelTransformRequest, with ModelRequestIdentifier as join key
+   */
+  private def buildModelTransformRequests(joinRequest: ResponseWithContext): Seq[ModelTransformRequest] = {
+    if (joinRequest.joinCodec.isEmpty) {
+      Seq.empty
+    } else {
+      val joinName = joinRequest.joinCodec.get.conf.join.metaData.name
+      val requests = joinRequest.joinCodec.get.conf.modelTransformsListScala.map { modelTransform =>
+        val modelName = modelTransform.model.metaData.name
+        val prefix = Option(modelTransform.prefix)
+        ModelTransformRequest(modelTransform,
+                              ModelRequestIdentifier(joinName, modelName, prefix),
+                              joinRequest.derivedValues)
+      }
```

Comment:
let's do de-dup here?
```suggestion
      }.toSet
      .toSeq
```